enigma/src/test/java/org/quiltmc/enigma/name_proposal/TestNameProposal.java

```diff
@@ -0,0 +1,105 @@
+package org.quiltmc.enigma.name_proposal;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.quiltmc.enigma.TestUtil;
+import org.quiltmc.enigma.api.Enigma;
+import org.quiltmc.enigma.api.EnigmaPlugin;
+import org.quiltmc.enigma.api.EnigmaPluginContext;
+import org.quiltmc.enigma.api.EnigmaProfile;
+import org.quiltmc.enigma.api.EnigmaProject;
+import org.quiltmc.enigma.api.ProgressListener;
+import org.quiltmc.enigma.api.analysis.index.jar.EntryIndex;
+import org.quiltmc.enigma.api.analysis.index.jar.JarIndex;
+import org.quiltmc.enigma.api.class_provider.ClasspathClassProvider;
+import org.quiltmc.enigma.api.service.NameProposalService;
+import org.quiltmc.enigma.api.source.TokenType;
+import org.quiltmc.enigma.api.translation.mapping.EntryMapping;
+import org.quiltmc.enigma.api.translation.mapping.EntryRemapper;
+import org.quiltmc.enigma.api.translation.representation.entry.Entry;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TestNameProposal {
+	private static final Path JAR = TestUtil.obfJar("enums");
+	private static EnigmaProject project;
+
+	@BeforeAll
+	public static void setupEnigma() throws IOException {
+		Reader r = new StringReader("""
+				{
+					"services": {
+						"name_proposal": [
+							{
+								"id": "test:name_all_fields_a"
+							},
+							{
+								"id": "test:name_all_fields_b"
+							},
+							{
+								"id": "test:name_all_fields_c"
+							},
+							{
+								"id": "test:name_all_fields_d"
+							}
```

Comment:
fun fact if you add a comma on this line it silently fails parsing and the test fails since it doesn't have any services to work with. we need to fix this!

Reply:
it doesn't throw an exception on `EnigmaProfile.parse()`?

Reply:
nothing, you can try it
I'll fix this myself though